alpha/declcfg/load.go

```diff
@@ -22,13 +22,74 @@ const (
 	indexIgnoreFilename = ".indexignore"
 )
 
+type WalkMetasFSFunc func(path string, meta *Meta, err error) error
+
+func WalkMetasFS(root fs.FS, walkFn WalkMetasFSFunc) error {
+	return walkFiles(root, func(root fs.FS, path string, err error) error {
+		if err != nil {
+			return walkFn(path, nil, err)
+		}
+
+		f, err := root.Open(path)
+		if err != nil {
+			return walkFn(path, nil, err)
+		}
+		defer f.Close()
+
+		return WalkMetasReader(f, func(meta *Meta, err error) error {
+			return walkFn(path, meta, err)
+		})
+	})
+}
+
+type WalkMetasReaderFunc func(meta *Meta, err error) error
+
+func WalkMetasReader(r io.Reader, walkFn WalkMetasReaderFunc) error {
+	dec := yaml.NewYAMLOrJSONDecoder(r, 4096)
+	for {
+		doc := json.RawMessage{}
+		if err := dec.Decode(&doc); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return walkFn(nil, err)
```

Comment:
Nit: Could we signal that the error occurred while attempting to decode like we do with the unmarshal error below? Could be useful for debugging.
```suggestion
			return walkFn(nil, fmt.Errorf("decoding error: %s", err))
```

Another thought, I could see it being useful to return a custom error type that the walkFn can use to easily determine what step the error occurred in (i.e a `DecodeError` and/or `UnmarshalError` types). Would be useful if the walkFn want's to perform some specific logic based on a specific error. That being said, maybe being that specific isn't useful in this case? I don't have any use cases for that off the top of my head.

Reply:
I could maybe see this:
```suggestion
			return walkFn(nil, fmt.Errorf("decoding error: %w", err))
```

I think it's important to give the walk function access to the underlying error if we wrap it with some context. In fact I was going to comment that in the case of the formatting we do for json unmarshaling errors, we might want to move that to a different spot so we don't obfuscate the original error.

Reply:
For now I think I'm just going to leave this as directly passing an unaltered error, which is what we were doing before. We can revisit if it turns out this extra context would be important in this case.